app/commands/v2/patch_link_set.rb

```diff
@@ -38,6 +32,14 @@ def call
 
     private
 
+      def find_or_create_link_set(content_id)
+        begin
+          LinkSet.find_or_create_by!(content_id: content_id).lock!
+        rescue ActiveRecord::RecordNotUnique
```

Comment:
@cbaines that's weird - I seem to have just got comments through from you that no doubt you wrote a number of days ago. Maybe they were caught in the github review process?